### ISSUE TITLE
REQUEST: Automatic Board Alignment

### ISSUE THREAD

**Issue by b14ckyy** [iNavFlight/inav#4355]
Hi Team,

As I'm short before the final tune flights of my Wing with an Mate F405 Wing FC, I recognized a point, that could be optimized. I need to tweak the board alignment setting after multiple flights till the acc. is perfectly level My FC is mounted 90° turned and a fixed wing always have a slight pitch angle. So atm I have set it to 270° sensor alignment and the board alignment to 4° pitch. Will test tomorrow if it works. 

Now my request: Would it be possible to have a auto board alignment tune mode on GPS enabled fixed wings? Or even copters? On a wing the Tuning could enable the Cruise mode and check if the plane is making a slight turn or gaining/loosing height. Then it could calculate an average pitch and roll value and save it as the board alignment value. This could be done multiple times for a few seconds. If the pilot leaves sticks, the measurement starts and is paused if the pilot has control to make a turn to stay in range. if the pilot does this multiple times in different flight directions, this should give you a very good alignment value. Of course this should be done at very calm wind conditions. 

The accellerometer Calibration is mostly done with the naked board and very hard to do when it is build inside the plane.

This Mode could also be combined with the servo trim mode to get the best tuning possible in one flight.

**Comment by Pairan** [comment]
the topic comes up here now and then ... :) And I like tha fact, that it isn't only me! Have a look at #2384 

Please write a comment on that there too ;)

**Comment by b14ckyy** [comment]
Thanks, that was not shown in the similar topics because it is closed. So Commenting does not make much sense I think :)
But yes I agree with you.

**Comment by Pairan** [comment]
simply respond to it and it is reactivated

**Comment by b14ckyy** [comment]
Today I found out, what a wrong board pitch tuning can cause :D 

My AR wing is set to 1400us cruise throttle. To check if this changes the loiter height fluctuation, i tested with 0° pitch alignment on the board. Then in cruise to keep height, INAV detected a 5-7° pitch up from straight flight and has then throttled up permanently to 1500. This caused an unnecessary high airspeed of 14m/s and 3.6A current draw on cruise mode. Generally this wing is totally fine with 10-11m/s at 1400 throttle and just pulls about 2.5-2.7A. So this makes Board alignment very important for an endurance or long range build and efficiency. Fortunately from this test today on very constant wind and long 2 way cruise I was able to get the best pitch alignment from the log. An automatic tune would be very helpful though :)
